dev_guide/builds.adoc

```diff
@@ -17,6 +17,68 @@ runnable images to be used on OpenShift. There are three types of builds:
 - STI build
 - custom build
 
+== Defining an incremental buildConfig
+STI builds provide an option to perform incremental build, which means they will
+reuse artifacts from previously built image. To create an incremental build one
+needs to create following BuildConfig:
```

Comment:
we should be able to avoid quoting the whole buildconfig again, since @mfojtik is adding a buildconfig example in this very same file.